src/main.ts

```diff
@@ -361,16 +366,18 @@ export default class ReadwisePlugin extends Plugin {
           }
           await this.fs.write(originalName, contentToSave);
           await this.removeBooksFromRefresh([bookID]);
+          await this.removeBookFromFailedBooks([bookID]);
         } catch (e) {
           console.log(`Readwise Official plugin: error writing ${processedFileName}:`, e);
           this.notice(`Readwise: error while writing ${processedFileName}: ${e}`, true, 4, true);
           if (bookID) {
             // handles case where user doesn't have `settings.refreshBooks` enabled
-            await this.addBookToRefresh(bookID);
+            await this.addToFailedBooks(bookID);

```

Comment:
to be safe, shouldn't this catch really only be for the `fs.write` call? it makes my head spin thinking a little bit about `this.removeBookFromFailedBooks` somehow failing, causing us to add that same book to the failedbooks again